File: Seccion 11 - Funciones en C++/Seccion_11_Dart/09_ejercicio_07.dart

class Wrapper {
  var value;

  Wrapper(this.value);
}

void intercambiar(Wrapper num1, Wrapper num2) {
  int aux;

  aux = num1.value;
  num1.value = num2.value;
  num2.value = aux;
}

void main() {

  Wrapper num1 = new Wrapper(10);
  Wrapper num2 = new Wrapper(15);

  print('El valor de num1.value es: ${num1.value}');
  print('El valor de num2.value es: ${num2.value}');

  intercambiar(num1, num2);

  print('\nEl nuevo valor de num1.value es: ${num1.value}');
  print('El nuevo valor de num2.value es: ${num2.value}');
}

File: Seccion 11 - Funciones en C++/Seccion_11_Python/09_ejercicio_07.py

num1 = 10
num2 = 15

def intercambiar():
    global num1
    global num2

    aux = num1
    num1 = num2
    num2 = aux

print('El valor de num1 es: {}'.format(num1))
print('El valor de num2 es: {}'.format(num2))

intercambiar()

print('\nEl nuevo valor de num1 es: {}'.format(num1))
print('El nuevo valor de num2 es: {}'.format(num2))